firewood/src/storage/linear/proposed.rs

```diff
@@ -164,51 +180,67 @@ impl<P: ReadLinearStore> WriteLinearStore for Proposed<P, Mutable> {
 // this is actually a clippy bug, works on nightly; see
 // https://github.com/rust-lang/rust-clippy/issues/12519
 #[allow(clippy::unused_io_amount)]
-use crate::storage::linear::layered::LayeredReader;
-
 #[cfg(test)]
 #[allow(clippy::unwrap_used)]
 mod test {
-    use super::super::tests::ConstBacked;
+    use crate::storage::linear::filebacked::tests::new_temp_filebacked;
+
     use super::*;
     use rand::Rng;
     use std::time::Instant;
     use test_case::test_case;
 
+    const TEST_DATA: &[u8] = b"random data";
+
     #[test]
     fn smoke_read() -> Result<(), std::io::Error> {
-        let sut: Proposed<ConstBacked, Immutable> = ConstBacked::new(ConstBacked::DATA).into();
+        let parent = new_temp_filebacked(TEST_DATA);
+
+        let proposed = Proposed::new(Arc::new(ImmutableLinearStore::FileBacked(parent)));
 
         // read all
-        let mut data = [0u8; ConstBacked::DATA.len()];
-        sut.stream_from(0).unwrap().read_exact(&mut data).unwrap();
-        assert_eq!(data, ConstBacked::DATA);
+        let mut data = [0u8; TEST_DATA.len()];
+        proposed
+            .stream_from(0)
+            .unwrap()
+            .read_exact(&mut data)
+            .unwrap();
+        assert_eq!(data, TEST_DATA);
 
         // starting not at beginning
-        let mut data = [0u8; ConstBacked::DATA.len()];
-        sut.stream_from(1).unwrap().read_exact(&mut data[1..])?;
-        assert_eq!(data.get(1..), ConstBacked::DATA.get(1..));
+        let mut data = [0u8; TEST_DATA.len()];
+        proposed
+            .stream_from(1)
+            .unwrap()
+            .read_exact(&mut data[1..])?;
+        assert_eq!(data.get(1..), TEST_DATA.get(1..));
 
         // ending not at end
-        let mut data = [0u8; ConstBacked::DATA.len() - 1];
-        sut.stream_from(0).unwrap().read_exact(&mut data)?;
-        assert_eq!(Some(data.as_slice()), ConstBacked::DATA.get(..data.len()));
+        let mut data = [0u8; TEST_DATA.len() - 1];
+        proposed.stream_from(0).unwrap().read_exact(&mut data)?;
+        assert_eq!(Some(data.as_slice()), TEST_DATA.get(..data.len()));
 
         Ok(())
     }
 
     #[test]
     fn smoke_mutate() -> Result<(), std::io::Error> {
-        let mut sut: Proposed<ConstBacked, Mutable> = ConstBacked::new(ConstBacked::DATA).into();
+        let parent = new_temp_filebacked(TEST_DATA);
```

Comment:
Not sure I love this, as it's now creating files whereas before it didn't need to. We aren't testing filebacked, we're just testing mutate.

Reply:
removed